rust-guest/src/context/context.rs

```diff
@@ -0,0 +1,102 @@
+#![allow(dead_code)]
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+
+use lazy_static::lazy_static;
+use std::sync::Mutex;
+
+use std::io::Cursor;
+use std::mem;
+use std::collections::HashMap;
+use super::generated::{Encode, Decode, Context, Request, Response};
+
+lazy_static! {
+    pub static ref PTR_LEN: Mutex<(u32, u32)> = Mutex::new((0, 0));
+}
+
+pub trait RunContext {
+    fn new() -> Self;
+    fn from_read_buffer(self, read_buff_global: &mut Cursor<&mut Vec<u8>>) -> Self;
+    fn to_write_buffer(self) -> (u32, u32);
```

Comment:
Can we define some of these functions as private? Is the default function necessary since `new` should technically handle this?

Reply:
Yes, and good point on default() -- totally redundant, removed.